Feature/game feat : 게임, 리뷰 기능 구현

89JHoon opened the pull request:
## #️⃣연관된 이슈

> 게임,리뷰 관련 전체적 기능

## 📝작업 내용

> 게임 CRUD
> 리뷰 CRUD

### 스크린샷 (선택)

## 💬리뷰 요구사항(선택)

> TODO> 권한 및 인증 부분 미 구현
> TODO> 첨부파일 수정 및 삭제 기능 추가 및 수정 필요

89JHoon committed:
init: 커밋 테스트

1. 커밋 테스트

89JHoon committed:
init: add 게임 등록 기본 CRUD 구현

1. 게임 등록 기능(기본)
2. 게임 조회(다건/단건) 기능(기본)
3. 게임 수정 기능(기본)
4. 게임 삭제 기능(기본)

89JHoon committed:
feat: add 게임 검색 기능 구현

1. 게임 검색 기능

89JHoon committed:
feat: add 리뷰 기능 구현

1. 리뷰 생성 기능(기본)
2. 리뷰 수정 기능(기본)
3. 리뷰 삭제 기능(기본)

89JHoon committed:
feat: add 게임과 리뷰 관련된  기능 구현

1. 게임 단건 조회시 리뷰 리스트 페이징으로 불러오기 구현
2. 게임 다건(검샘포함) 평균별점, 리뷰 수 불러오기 구현

89JHoon committed:
feat: add 게임등록 요청 관련된  기능 구현

1. 게임등록 요청
2. 게임등록 요청 삭제
3. 게임등록 요청 수정
4. 게임등록 요청 조회

89JHoon committed:
refactor: add 파일 구조 변경

1. 도메인.game 패키지 추가로 인한 파일 이동

89JHoon committed:
refactor: add 파일 구조 변경

1. 도메인.review 패키지 추가로 인한 파일 이동

89JHoon committed:
feat: add 게임 전체 조회 , 검색 코드 통합 /games

1. 게임 전체 조회 , 검색 코드 통합 /games 로 통합하여 사용

89JHoon committed:
feat: add 게임 이미지 첨부 파일 기능 관련 구형

1. 게임 등록시 이미지 첨부 파일  기능 구현
2. 게임 삭제시 이미지 데이터 삭제 기능 구현

89JHoon committed:
refact: 코드 정리 1차

1. PR 전 간단한 코드 정리

89JHoon committed:
Merge branch 'develop' into feature/game

89JHoon committed:
feat: 게임 첨부 파일 업로드 기능 구현

1. 게임 등록 첨부 파일 등록 기능

Newbiekk-kkh commented on src/main/java/com/example/gamemate/domain/game/controller/GameController.java:
```diff
@@ -0,0 +1,121 @@
+public class GameController {
+    private final GameService gameService;
+
+    @Autowired
```
@ AutoWired 말고 @ RequiredArgsConstructor 로 통일하면 좋을 것 같습니다.

89JHoon replied:
넵 수정하였습니다

Newbiekk-kkh commented on src/main/java/com/example/gamemate/domain/game/controller/GameController.java:
```diff
@@ -0,0 +1,121 @@
+        }
+
+        GameCreateResponseDto responseDto = gameService.createGame(requestDto, file);
+        return ResponseEntity.status(HttpStatus.CREATED).body(responseDto);
```
return 값을 new ResponseEntity<>(); 로 통일하기로 코드 컨벤션에 적혀있어서 통일하면 좋을 것 같습니다.

89JHoon replied:
넵 수정하였습니다!

Newbiekk-kkh commented on src/main/java/com/example/gamemate/domain/game/controller/GameController.java:
```diff
@@ -0,0 +1,121 @@
+     * @return
+     */
+    @PatchMapping("/{id}")
+    public ResponseEntity<GameUpdateResponseDto> updateGame(@PathVariable Long id,
```
반환 타입 void 로 고치면 좋을 것 같습니다.

89JHoon replied:
넵 수정하였습니다

Newbiekk-kkh commented on src/main/java/com/example/gamemate/domain/game/controller/GameEnrollRequestController.java:
```diff
@@ -0,0 +1,80 @@
+public class GameEnrollRequestController {
+    private final GameEnrollRequestService gameEnrollRequestService;
+
+    @Autowired
```
@ AutoWired 말고 @ RequiredArgsConstructor 로 통일하면 좋을 것 같습니다.

89JHoon replied:
넵 수정하였습니다

Newbiekk-kkh commented on src/main/java/com/example/gamemate/domain/game/controller/GameEnrollRequestController.java:
```diff
@@ -0,0 +1,80 @@
+     * @return
+     */
+    @PatchMapping("/{id}")
+    public ResponseEntity<GameEnrollRequestResponseDto> updateGameEnroll(@PathVariable Long id, @RequestBody GameEnrollRequestUpdateRequestDto requestDto) {
```
반환타입 void 로 고치면 좋을것같습니다.

89JHoon replied:
넵 수정하였습니다

Newbiekk-kkh commented on src/main/java/com/example/gamemate/domain/game/service/GameEnrollRequestService.java:
```diff
@@ -0,0 +1,99 @@
+        this.gameEnrollRequestRepository = gameEnrollRequestRepository;
+    }
+
+    public GameEnrollRequestResponseDto createGameEnrollRequest(GameEnrollRequestCreateRequestDto requestDto) {
```
@ Transactional 붙이면 좋을것같습니다.

Newbiekk-kkh commented on src/main/java/com/example/gamemate/domain/game/service/GameEnrollRequestService.java:
```diff
@@ -0,0 +1,99 @@
+        return gameEnrollRequestRepository.findAll(pageable).map(GameEnrollRequestResponseDto::new);
+    }
+
+    @Transactional
```
@ Transactional (readOnly = true) 로 변경하면 좋을것같습니다.

Newbiekk-kkh commented on src/main/java/com/example/gamemate/domain/game/service/GameEnrollRequestService.java:
```diff
@@ -0,0 +1,99 @@
+        }
+    }
+
+    public void deleteGame(Long id) {
```
트랜잭션 처리 해주시면 좋을것같습니다.

Newbiekk-kkh commented on src/main/java/com/example/gamemate/domain/game/service/GameService.java:
```diff
@@ -0,0 +1,167 @@
+        this.gameImageRepository=gameImageRepository;
+    }
+
+    public GameCreateResponseDto createGame(GameCreateRequestDto gameCreateRequestDto , MultipartFile file) {
```
트랜잭션 처리해주시면 좋을것 같습니다.

89JHoon replied:
넵 수정하였습니다

Newbiekk-kkh commented on src/main/java/com/example/gamemate/domain/review/controller/ReviewController.java:
```diff
@@ -0,0 +1,67 @@
+     * @param requestDto
+     * @return
+     */
+    @PatchMapping("/{id}")
```
반환타입 void로 바꾸면 좋을것같습니다.

89JHoon replied:
넵 수정하였습니다

Newbiekk-kkh commented on src/main/java/com/example/gamemate/domain/review/entity/Review.java:
```diff
@@ -0,0 +1,45 @@
+@Getter
+@NoArgsConstructor
+@Table(name = "review")
+public class Review extends com.example.gamemate.global.common.BaseEntity {
```
경로 설정으로 되어있어서 깔끔하게 써주시면 좋을것같습니다.

89JHoon replied:
넵 수정하였습니다

Newbiekk-kkh commented on src/main/java/com/example/gamemate/domain/review/service/ReviewService.java:
```diff
@@ -0,0 +1,68 @@
+        this.gameRepository = gameRepository;
+    }
+
+    public ReviewCreateResponseDto createReview(Long gameId, ReviewCreateRequestDto requestDto) {
```
트랜잭션 처리하면 좋을것같습니다.

89JHoon replied:
넵 수정하였습니다

Newbiekk-kkh commented on src/main/java/com/example/gamemate/domain/review/service/ReviewService.java:
```diff
@@ -0,0 +1,68 @@
+        return new ReviewCreateResponseDto(saveReview);
+    }
+
+    public ReviewUpdateResponseDto updateReview(Long gameId, Long id, ReviewUpdateRequestDto requestDto) {
```
반환 void로 하면 좋을것같습니다.

89JHoon replied:
넵 수정하였습니다

sumyeom committed:
Merge branch 'develop' into feature/game

yeji-world commented on src/main/java/com/example/gamemate/domain/game/controller/GameController.java:
```diff
@@ -0,0 +1,121 @@
+     * @return
+     */
+    @PostMapping(consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
+    @ResponseStatus(HttpStatus.CREATED)
```
아래 return에서 명시되어있어서 빼도 될 것 같아요~

89JHoon replied:
넵 수정하였습니다

yeji-world commented on src/main/java/com/example/gamemate/domain/game/controller/GameController.java:
```diff
@@ -0,0 +1,121 @@
+     * 게임 전체 조회
+     *
+     * @param page
+     * @param szie
```
여기 오타가 있네용
szie -> size 로 수정 부탁드립니다~!
혹시 @param을 일부만 적으신 이유가 있을까요?

89JHoon replied:
아직 자바독을 완성하지 못했습니다! ㅠㅠ

yeji-world commented on src/main/java/com/example/gamemate/domain/game/controller/GameController.java:
```diff
@@ -0,0 +1,121 @@
+     * 게임 정보 수정
+     *
+     * @param id
+     * @param requestDto
```
이 부분 수정이 필요할 것 같습니다~!

89JHoon replied:
넵 수정하겟습니다!

89JHoon committed:
refact: 컨트롤러 파라미터 줄 바꿈 수정

1. 컨트롤러 메서드에서 파라미터 줄바꿀 때 ( 뒤에서 엔터!

89JHoon committed:
Merge branch 'develop' into feature/game

# Conflicts:
#	src/main/java/com/example/gamemate/global/constant/ErrorCode.java

89JHoon committed:
Merge branch 'origin/feature/game' into feature/game

89JHoon committed:
Add .DS_Store to .gitignore

89JHoon committed:
refact: final 생성자는 @RequiredArgsConstructor 설정

1. final 생성자는 @RequiredArgsConstructor 설정

Newbiekk-kkh reviewed (commented):
확인부탁드립니다

yeji-world commented on src/main/java/com/example/gamemate/domain/game/controller/GameEnrollRequestController.java:
```diff
@@ -0,0 +1,86 @@
+    public ResponseEntity<Page<GameEnrollRequestResponseDto>> findAllGameEnrollRequest() {
+
+        Page<GameEnrollRequestResponseDto> gameEnrollRequestAll = gameEnrollRequestService.findAllGameEnrollRequest();
+        return ResponseEntity.ok(gameEnrollRequestAll);
```
return 값을 new ResponseEntity<>(); 로 수정 부탁드립니다~!

89JHoon replied:
네 수정하였습니다!

yeji-world commented on src/main/java/com/example/gamemate/domain/game/controller/GameEnrollRequestController.java:
```diff
@@ -0,0 +1,86 @@
+            @PathVariable Long id) {
+
+        GameEnrollRequestResponseDto gameEnrollRequestById = gameEnrollRequestService.findGameEnrollRequestById(id);
+        return ResponseEntity.ok(gameEnrollRequestById);
```
return 값을 new ResponseEntity<>(); 로 수정 부탁드립니다

89JHoon replied:
네 수정하였습니다!

yeji-world commented on src/main/java/com/example/gamemate/domain/game/dto/GameEnrollRequestUpdateRequestDto.java:
```diff
@@ -0,0 +1,23 @@
+    private Boolean isAccepted;
+
+
+    public GameEnrollRequestUpdateRequestDto(String title, String genre, String platform , String description,Boolean isAccepted  ) {
```
public GameEnrollRequestUpdateRequestDto(String title, String genre, String platform, String description, Boolean isAccepted) {
 띄어쓰기 수정 부탁드립니다

yeji-world commented on src/main/java/com/example/gamemate/domain/game/dto/GameUpdateRequestDto.java:
```diff
@@ -0,0 +1,21 @@
+    private String description;
+
+
+    public GameUpdateRequestDto(String title, String genre, String platform , String description) {
```
public GameUpdateRequestDto(String title, String genre, String platform, String description) {
띄어쓰기 수정 부탁드립니다

89JHoon replied:
네 수정하였습니다!

yeji-world commented on src/main/java/com/example/gamemate/domain/game/entity/Game.java:
```diff
@@ -0,0 +1,69 @@
+@Entity
+@Getter
+@Table(name = "game")
+@AllArgsConstructor
```
@AllArgsConstructor를 제한없이 사용하는 것에 대해 같이 고민해보면 좋을 것 같아요
(access = AccessLevel.PRIVATE)

89JHoon replied:
네! 확인해보고 수정하겟습니다!

yeji-world commented on src/main/java/com/example/gamemate/domain/game/entity/GamaEnrollRequest.java:
```diff
@@ -0,0 +1,54 @@
+@Entity
+@Getter
+@Table(name = "game_enroll_request")
+@AllArgsConstructor
```
@AllArgsConstructor를 제한없이 사용하는 것에 대해 같이 고민해보면 좋을 것 같아요
(access = AccessLevel.PRIVATE)

89JHoon replied:
네! 확인해보고 수정하겟습니다!

yeji-world commented on src/main/java/com/example/gamemate/domain/review/controller/ReviewController.java:
```diff
@@ -0,0 +1,75 @@
+
+    private final ReviewService reviewService;
+
+    @Autowired
```
@AutoWired 대신 @RequiredArgsConstructor 로 통일하면 좋을 것 같습니다.

89JHoon replied:
넵 수정하였습니다

yeji-world commented on src/main/java/com/example/gamemate/domain/review/controller/ReviewController.java:
```diff
@@ -0,0 +1,75 @@
+            @RequestBody ReviewCreateRequestDto requestDto) {
+
+        ReviewCreateResponseDto responseDto = reviewService.createReview(gameId, requestDto);
+        return ResponseEntity.status(HttpStatus.CREATED).body(responseDto);
```
return 값을 new ResponseEntity<>(); 로 수정 부탁드립니다

89JHoon replied:
네  수정하겟습니다!

89JHoon committed:
refact: game feat : 게임, 리뷰 기능 구현 #25 PR코드수정

1. @AutoWired 말고 @ RequiredArgsConstructor 로 통일
2. 오타 및 줄간격 경로설정 수정
3. return 값을 new ResponseEntity<>(); 로 통일
4. 반환 타입 void 로 수정
5. 트랜잭션 처리
6. @AllArgsConstructor를 제한없이 사용하는 것 > 삭제

sumyeom commented on src/main/java/com/example/gamemate/domain/game/controller/GameController.java:
```diff
@@ -0,0 +1,125 @@
+        }
+
+
+        return ResponseEntity.ok(games);
```
이 부분도return new ResponseEntity<>(games, HttpStatus.OK);
이렇게 해야될 것 같아요!

sumyeom commented on src/main/java/com/example/gamemate/domain/game/controller/GameController.java:
```diff
@@ -0,0 +1,125 @@
+            @PathVariable Long id) {
+
+        GameFindByIdResponseDto gameById = gameService.findGameById(id);
+        return ResponseEntity.ok(gameById);
```
이 부분도 return new ResponseEntity<>(gameById, HttpStatus.OK);
이렇게 해야될 것 같아요!

sumyeom commented on src/main/java/com/example/gamemate/domain/game/controller/GameEnrollRequestController.java:
```diff
@@ -0,0 +1,87 @@
+            @RequestBody GameEnrollRequestCreateRequestDto requestDto) {
+
+        GameEnrollRequestResponseDto responseDto = gameEnrollRequestService.createGameEnrollRequest(requestDto);
+        return ResponseEntity.status(HttpStatus.CREATED).body(responseDto);
```
이 부분도 return new ResponseEntity<>( ) 로 수정되어야할 것 같습니다

89JHoon committed:
refact: game feat : 게임, 리뷰 기능 구현 #25 PR코드수정

1. return 값을 new ResponseEntity<>(); 로 통일
